crates/primitives-traits/src/signature.rs

```diff
@@ -0,0 +1,43 @@
+//! Signature abstraction
+
+use alloy_primitives::{Address, Parity, B256, U256};
+
+/// Reth extension for alloy type [`Signature`](alloy_primitives::Signature).
+pub trait Signature: Sized + Send + Sync {
+    /// Returns ref to `r` value.
+    fn r(&self) -> U256;
+
+    /// Returns ref to `s` value.
+    fn s(&self) -> U256;
+
+    /// Returns ref to `v` value.
+    fn v(&self) -> Parity;
```

Comment:
where do we need these? my general question is, why is this not just an associated type `Signature` with at most a way to recover the signer (like in alloy), are there specific purposes for the rest of the fns?

Reply:
the v, r, s getters were used for default trait method impls, see commit [1532905](https://github.com/paradigmxyz/reth/pull/11567/commits/1532905ce17fcfdc56f17ddb991345ed42a6f937)

they are all taken from this file, in order to separate the feature gated op logic out of it 
https://github.com/paradigmxyz/reth/blob/main/crates/primitives/src/transaction/signature.rs

Reply:
I don't see how this solves for op given the deposit tx has no signature, and these are therefore not implementable for that "signature" type. I think this would be better as an associated type like in op? or perhaps as Arsenii suggested, transactions should know how to decode themselves

Reply:
it seems to me like we need wrappers rather than impl this for every tx type redundantly
```rust
pub enum Signature {
    NonEip155(alloy_primitives::Signature)
    Eip155(alloy_primitives::Signature)
}
```
and a trait that has trait methods for en-/decoding, extract chain id and recover signers. the op type to impl the trait would be
```rust
pub enum OpSignature {
    Eth(Signature)
    System(alloy_primtives::Signature)
}
```
possibly recover signers can be replaced in favour of using alloy methods https://github.com/alloy-rs/core/blob/d9ad5fb4e6aabdaefd174e8397e747c00a59efbb/crates/primitives/src/signature/sig.rs#L192-L246 ?